zerver/lib/user_groups.py

```diff
@@ -378,13 +378,15 @@ def get_subgroup_ids(user_group: UserGroup, *, direct_subgroup_only: bool = Fals
 
 def get_recursive_subgroups_for_groups(
     user_group_ids: Iterable[int], realm: Realm
-) -> QuerySet[UserGroup]:
+) -> QuerySet[NamedUserGroup]:
```

Comment:
We use this function only for updating the subgroups and deleting the groups. So, the `user_group_ids` would always be of `NamedUserGroup` and we would anyways not allow to set `UserGroup` objects as subgroups, so this would also only return the `NamedUserGroup` objects.

Reply:
Yeah I think returning the `NamedUserGroup` type here will be most convenient for client code too.